### PR TITLE
Add flag to enable beta features

### PR DESCRIPTION
**Pull request by pkrull-ansys**

### CODE REVIEW

**Review by safeerehman — APPROVED** [review]

File: examples/12_additive_3d_microstructure_beta.py [hunk 1]
```diff
@@ -60,7 +60,8 @@
 
 from ansys.additive.core import Additive, AdditiveMachine, Microstructure3DInput, SimulationError
 
-additive = Additive()
+# Beta features must be enable to use the 3D microstructure simulation.
```
Comment (julieatansys) [inline examples/12_additive_3d_microstructure_beta.py:63]:
```suggestion
# Beta features must be enabled to use the 3D microstructure simulation.
```

File: src/ansys/additive/core/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Provides custom exceptions for the additive client."""
+
+
+class BetaFeatureNotEnabledError(Exception):
+    """Raised when a beta features is not enabled."""
```
Comment (julieatansys) [inline src/ansys/additive/core/exceptions.py:26]:
```suggestion
    """Raised when beta features are not enabled."""
```

File: src/ansys/additive/core/additive.py [hunk 3]
```diff
@@ -213,6 +218,16 @@ def nsims_per_server(self, value: int) -> None:
...
         self._nsims_per_server = value
 
+    @property
+    def enable_beta_features(self) -> bool:
+        """Flag indicating if beta features are enabled."""
+        return self._enable_beta_features
+
+    @enable_beta_features.setter
+    def enable_beta_features(self, value: bool) -> None:
+        """Set the flag indicating if beta features are enabled."""
```
Comment (julieatansys) [inline src/ansys/additive/core/additive.py:228]: You could say: Set the flag to enable beta features.
(But I'm not quite sure what usage you intend here.)

File: src/ansys/additive/core/additive.py [hunk 4]
```diff
@@ -332,6 +354,12 @@ def _simulate(
         if input.material == AdditiveMaterial():
             raise ValueError("A material is not assigned to the simulation input")
 
+        if isinstance(input, Microstructure3DInput) and self.enable_beta_features is False:
+            raise BetaFeatureNotEnabledError(
+                "3D microstructure simulation requires beta features to be enabled.\n"
```
Comment (julieatansys) [inline src/ansys/additive/core/additive.py:359]:
```suggestion
                "3D microstructure simulation requires that beta features be enabled.\n"
```

**Review by julieatansys — COMMENTED** [review]
See my suggestions.

File: src/ansys/additive/core/additive.py (hunk 3 above)
Comment (pkrull-ansys) [reply src/ansys/additive/core/additive.py:228]: This can be "set" to True or False. Setting it to False disables the beta features.

Comment (julieatansys) [reply src/ansys/additive/core/additive.py:228]: ok

**Pull request merged**